### ISSUE TITLE
Handle/fix "Got unexpected Kafka message" error

### ISSUE THREAD

**Issue by IgorFedchenko** [akkadotnet/Akka.Streams.Kafka#99]
During test execution under Ubuntu:
Test:
`Akka.Streams.Kafka.Tests.Integration.SourceWithOffsetContextIntegrationTests.SourceWithOffsetContext_at_least_once_consuming_should_work`

Error:
```
Failed: Expected a message of type Akka.Streams.TestKit.TestSubscriber+OnNext1[System.ValueTuple2[Akka.NotUsed,Akka.Streams.Kafka.Messages.ICommittableOffsetBatch]], but received {TestSubscriber.OnError(Got unexpected Kafka message: {"Topic":"topic-1-3aac2d51-7efe-4e0c-adca-0bac512289e9","Partition":{"Value":0,"IsSpecial":false},"Offset":{"Value":0,"IsSpecial":false},"TopicPartition":{"Topic":"topic-1-3aac2d51-7efe-4e0c-adca-0bac512289e9","Partition":{"Value":0,"IsSpecial":false}},"TopicPartitionOffset":{"Topic":"topic-1-3aac2d51-7efe-4e0c-adca-0bac512289e9","Partition":{"Value":0,"IsSpecial":false},"Offset":{"Value":0,"IsSpecial":false},"TopicPartition":{"Topic":"topic-1-3aac2d51-7efe-4e0c-adca-0bac512289e9","Partition":{"Value":0,"IsSpecial":false}}},"Message":{"Value":"3","Timestamp":{"Type":1,"UnixTimestampMs":1580383308079,"UtcDateTime":"2020-01-30T11:21:48.079Z"},"Headers":[]},"Value":"3","Timestamp":{"Type":1,"UnixTimestampMs":1580383308079,"UtcDateTime":"2020-01-30T11:21:48.079Z"},"Headers":[],"IsPartitionEOF":false})} (type Akka.Streams.TestKit.TestSubscriber+OnError) instead from [akka://SourceWithOffsetContextIntegrationTests/user/StreamSupervisor-27/Flow-1-0-unknown-operation#1937253766]

Expected: True

Actual: False
```

Details: https://dev.azure.com/dotnet/Akka.NET/_build/results?buildId=27932&view=ms.vss-test-web.build-test-results-tab&runId=1085592&resultId=100006&paneView=debug

It is thrown from `KafkaConsumerActor`:
https://github.com/akkadotnet/Akka.Streams.Kafka/blob/875b806c105cdd3c4cdfc2cb0b6c570370331340/src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs#L304-L311

Basically partition should be paused, but for some reason message is still received.

I think that instead of throwning exception here, we can log warning and make a workaround - ignore message and reset partition position pointer by 1 .

**Comment by Aaronontheweb** [comment]
Fixed - this was related to calling `Seek` / `Pause` at times when the `Consumer<K,V>` can't support it, which is not particularly well-documented inside the Confluent documentation.
